.github/workflows/action-build.yml

```diff
@@ -26,22 +26,26 @@ jobs:
         run: |
           gh pr checkout ${{ github.event.pull_request.number }}
 
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 7
+
       - uses: actions/setup-node@v3
         with:
           node-version: '16'
-          cache: 'yarn'
+          cache: 'pnpm'
 
-      - run: yarn install --frozen-lockfile
+      - run: pnpm install --frozen-lockfile
 
-      - run: yarn build
+      - run: pnpm build
 
       - name: Commit and push if needed
         run: |
-          git add .
-          if [ -z "$(git status --porcelain)" ]; then
-            echo "💾 no changes to dist/index.js"
+          if [ "$(git diff --ignore-space-at-eol | wc -l)" -eq "0" ]; then
+            echo "❎ no changes"
```

Comment:
I would use ℹ️ instead of a (green) cross which would make me think twice before deciding if it's all good.